CMakeLists.txt

```diff
@@ -427,6 +432,9 @@ else()
     set(INSTALL_TARGETS xgboost)
   endif()
 endif()
+if(BUILD_WITH_EXTERNAL_CCCL)
+  list(APPEND INSTALL_TARGETS _CUB_CUB _CUB_libcudacxx)
```

Comment:
Do we need `libcudacxx` in `find_package` as well?

Reply:
We are not using it at the moment.

Reply:
Interesting ok. So in what way is `libcudacxx` being used on this line?

Reply:
This is an artifact of the install script. Without this line, CMake complains:
```
CMake Error: install(EXPORT "XGBoostTargets" ...) includes target "_CUB_libcudacxx" which requires target "_Thrust_libcudacxx" that is not in any export set.
```

Reply:
Think we may want to use `target_link_libraries` to accomplish this

For [example]( https://github.com/NVIDIA/cccl/blob/1b16af7ccc2f3aeb4c94a743c951bae3e0586052/examples/example_project/README.md?plain=1#L39-L41 ):

> ```cmake
> # Alternatively, if you're building a library
> add_library(your_library your_file.cu)
> target_link_libraries(your_library PRIVATE CCCL::CCCL)
> ```

Alternatively [with Thrust]( https://github.com/NVIDIA/cccl/blob/1b16af7ccc2f3aeb4c94a743c951bae3e0586052/thrust/thrust/cmake/README.md?plain=1#L20-L22 ) (note `target_link_libraries` probably should use `private` here too):

```cmake
find_package(Thrust REQUIRED CONFIG)
thrust_create_target(Thrust)
target_link_libraries(MyProgram Thrust)
```

Reply:
Hmm, somehow with the latest changes, I don't need to add install targets to avoid CMake error. Problem solved, I suppose?

Reply:
Was wondering if that might be the case. Great, thanks Hyunsu! 🙏